build.sbt

```
@@ -656,6 +723,29 @@ lazy val `java-shopping-cart` = (project in file("samples/java-shopping-cart"))
     }
   )
 
+lazy val `scala-shopping-cart` = (project in file("samples/scala-shopping-cart"))
+  .dependsOn(`scala-support`)
+  .enablePlugins(AkkaGrpcPlugin)
+  .settings(
+    name := "scala-shopping-cart",
+    PB.generate in Compile := (PB.generate in Compile).dependsOn(PB.generate in (`scala-support`, Compile)).value,
+    PB.protoSources in Compile ++= {
+      val baseDir = (baseDirectory in ThisBuild).value / "protocols"
+      Seq(baseDir / "frontend", baseDir / "example")
```

@janory Have you tried

```suggestion
      Seq(baseDir / "example")
```

---

@viktorklang 
I've checked out the branch & run the TCK on a different computer and it worked well, so I will cleanup the project and try it again. It is definitely a local issue.

The TCK is almost complete, the only issue is this:
```
[info] - must verify that the HTTP API of ShoppingCart protocol works *** FAILED ***
"...ems":[{"productId":"[B14623482","name":"Basic","quantity":1},{"productId":"A14362347","name":"Deluxe","quantity":7]}]}" did not equal "...ems":[{"productId":"[A14362347","name":"Deluxe","quantity":7},{"productId":"B14623482","name":"Basic","quantity":1]}]}" 
```

It is the same content, but the order is different. 
I will check the implementation and fix the order, although maybe we can relax the TCK's check to ignore the order and focus only the content, because my guess is, that the order here is not important.

---

@janory I had the excactly same "issue" in the Go Support first :-), had the items in a list that is not sorted.

---

@marcellanz :)
It's a one liner fix, I am just not sure if we really need to enforce the order here.

---

Had the same thought, yes.
There are other implicit requirements by the shopping cart tck example. Adding negative amounts of line-items as an example. As the shopping cart is used to verify the spec through the TCK, we perhaps might document these requirements within the shoppingcart.proto file.

---

The reason for checking the response as a String was to avoid having parsing influencing things needlessly. I'm very much open for a better way of verifying the responses :)